ref_kernels/ind/bli_gemmtrsm1m_ref.c

```diff
@@ -59,149 +449,79 @@ void PASTEMAC3(ch,opname,arch,suf) \
+	const inc_t       cs_b        = bli_trsm_var_cntl_nr_bcast( params ); \
 \
-	const inc_t       rs_b        = packnr; \
-	const inc_t       cs_b        = bli_cntx_get_blksz_def_dt( dt_r, BLIS_BBN, cntx ); \
-\
-	const ctype_r*    zero_r      = PASTEMAC(chr,0); \
-	const ctype_r*    minus_one_r = PASTEMAC(chr,m1); \
+	ctype_r* restrict zero_r      = PASTEMAC(chr,0); \
+	ctype_r* restrict minus_one_r = PASTEMAC(chr,m1); \
 \
 	const ctype_r     alpha_r     = PASTEMAC(ch,real)( *alpha ); \
-	const ctype_r     alpha_i     = PASTEMAC(ch,imag)( *alpha ); \
-\
-	      ctype_r*    b_use; \
-\
-	      inc_t       rs_b_use; \
-	      inc_t       cs_b_use; \
-\
-	      ctype       ct[ BLIS_STACK_BUF_MAX_SIZE \
-	                      / sizeof( ctype ) ] \
-	                      __attribute__((aligned(BLIS_STACK_BUF_ALIGN_SIZE))); \
-	/* FGVZ: Should we be querying the preference of BLIS_GEMMTRSM_?_UKR
-	   instead? */ \
-	const bool        col_pref    = bli_cntx_ukr_prefers_cols_dt( dt, BLIS_GEMM_VIR_UKR, cntx ); \
-	const inc_t       rs_ct       = ( col_pref ? 1 : nr ); \
-	const inc_t       cs_ct       = ( col_pref ? mr : 1 ); \
-\
-	const bool        use_ct      = ( m < mr || n < nr ); \
-\
-	      ctype*      c11_use     = c11; \
-\
-	      inc_t       rs_c_use    = rs_c; \
-	      inc_t       cs_c_use    = cs_c; \
-\
-	if ( use_ct ) \
-	{ \
-		c11_use  = ct; \
-		rs_c_use = rs_ct; \
-		cs_c_use = cs_ct; \
-	} \
-\
-\
-	/* Handle alphas with non-zero imaginary components. */ \
-	/* NOTE: This branch should never execute because alphas with
-	   non-zero imaginary components should be applied during
-	   packing, and so the only alphas we should see here are
-	   those exclusively in the real domain, either because the
-	   value originally had no imaginary compoent (e.g. 4.0) or
-	   because a 1.0 was sent in as a placeholder since the alpha
-	   was applied during packing. */ \
-	if ( 0 ) \
-	if ( !PASTEMAC(chr,eq0)( alpha_i ) ) \
-	{ \
-		bli_abort(); \
-\
-		/*
-		ctype_r* restrict one_r = PASTEMAC(chr,1); \
-\
-		const inc_t ld_b = rs_b; \
-\
-		PASTEMAC(ch,scal1ms_mxn)( schema_b, \
-		                          mr, \
-		                          nr, \
-		                          alpha, \
-		                          b11, rs_b, cs_b, ld_b ); \
-\
-		alpha_r = *one_r; \
-		*/ \
-	} \
+	const ctype_r     alpha_i     = PASTEMAC(ch,imag)( *alpha ); ( void ) alpha_i; \
 \
-\
-	{ \
-		/* Set the strides for the temporary bt matrix based on the native
-		   real domain micro-kernel storage preferences. */ \
-		if ( col_pref_r ) { rs_bt   = 1;    cs_bt   = mr;     \
-		                    rs_bt_r = 1;    cs_bt_r = mr_r; } \
-		else              { rs_bt   = nr;   cs_bt   = 1;      \
-		                    rs_bt_r = nr_r; cs_bt_r = 1;    } \
-\
-		b_use    = ( ctype_r* )bt; \
-		rs_b_use = rs_bt_r; \
-		cs_b_use = cs_bt_r; \
-	} \
+	auxinfo_t auxinfo_r = *auxinfo; \
+    bli_auxinfo_set_params( params_r, &auxinfo_r ); \
```

Comment:
Can you briefly remind me why the `cntl_t` nodes now track both a `params` and a `real_params` field? Is this just for supporting 1m, or might it have other applications?